stack/stack_funcs.cpp

```diff
@@ -0,0 +1,78 @@
+#include "stack_funcs.h"
+
+void stack_ctor(my_stack *stk) {
+    stk->canary_start = CANARY_VALUE;
+    stk->canary_end = CANARY_VALUE;
+    stk->data = (stackelem_t *)calloc(2, sizeof(stackelem_t));
+    stk->data[0] = CANARY_VALUE;
+    stk->size = 1; // 1 = 0
+    stk->capacity = reall_factor;
+    stk->data[stk->capacity - 1] = CANARY_VALUE;
+    stk->popped = NAN;
+    stk->hash = hash_struct(stk);
+    printf("ctor!\n");
+    stack_dump(stk); 
+}
+
+my_stack *realloc_incr_stack(my_stack *stk) {
```

Comment:
you should call stack_verify in every function that can be called outside. also, this function should be static. to prevent user from using it

Reply:
resolved